dashboard/app/controllers/featured_projects_controller.rb

```diff
@@ -1,23 +1,19 @@
 class FeaturedProjectsController < ApplicationController
-  def create
-    unless current_user.try(:permission?, UserPermission::PROJECT_VALIDATOR)
-      _, channel_id = storage_decrypt_channel_id(featured_project_params[:project_id])
-      @featured_project = FeaturedProject.create({storage_app_id: channel_id, who_featured_user_id: current_user.id})
-      @featured_project.save!
-    end
-  end
+  authorize_resource
 
-  def destroy_by_project_id
-    unless current_user.try(:permission?, UserPermission::PROJECT_VALIDATOR)
-      _, channel_id = storage_decrypt_channel_id(params[:project_id])
-      @featured_project = FeaturedProject.find_by_storage_app_id(channel_id)
-      @featured_project.destroy
-    end
+  def feature
+    _, channel_id = storage_decrypt_channel_id(params[:project_id])
+    return render_404 unless channel_id
+    @featured_project = FeaturedProject.find_or_create_by(storage_app_id: channel_id)
+    @featured_project.update_attribute(:unfeatured_at, nil)
+    @featured_project.update_attribute(:featured_at, DateTime.now)
+    @featured_project.save!
```

Comment:
These should be condensed to
```ruby
@featured_project.update! unfeatured_at: nil, featured_at: DateTime.now
```

Also as it stands, the save is redundant. I don't think [update_attribute](https://apidock.com/rails/ActiveRecord/Persistence/update_attribute) does what you think. See below

Reply:
#20448 ✅